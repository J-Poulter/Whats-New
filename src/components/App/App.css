* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.app {
  display: flex;
  background: #141527;
  height: 100vh;
  text-align: center;
  width: 100%;
}

.leftColumn {
  width: 20%;
  color: white;
  position: sticky;
}

.rightColumn {
  width: 80%;
  overflow-y: scroll;
}