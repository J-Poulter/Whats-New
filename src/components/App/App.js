import React, { Component } from 'react';
import entertainment from '../../data/entertainment';
import health from '../../data/health';
import local from '../../data/local';
import science from '../../data/science';
import technology from '../../data/technology';

import Menu from '../Menu/Menu';
import SearchForm from '../SearchForm/SearchForm';
import NewsContainer from '../NewsContainer/NewsContainer';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = { category: local }
    this.categories = [
      { title: 'entertainment', dataset: entertainment, iconImg: 'â˜…' }, 
      { title: 'local', dataset: local, iconImg: 'ðŸ’¡' }, 
      { title: 'health', dataset: health, iconImg: 'ðŸŽ¦' }, 
      { title: 'science', dataset: science, iconImg: 'ðŸ”¬' }, 
      { title: 'technology', dataset: technology, iconImg: 'ðŸŒ¡' }
    ]
  }


  selectCategory = (categoryTopic) => {
    this.setState({category: categoryTopic})
  }

  render () {
    return (
      <div className='app'>
        <div className='leftColumn'>
          <Menu categories={this.categories}
            selectCategory={this.selectCategory}/>
        </div>
        <div className='rightColumn'>
          <SearchForm category={this.state.category}
            selectCategory={this.selectCategory}/>
          <NewsContainer category={this.state.category}/>
        </div>
      </div>
    );
  }
}

export default App;
