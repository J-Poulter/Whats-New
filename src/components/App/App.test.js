import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render all components needed to match the snapshot', () => {
    const snap = render(<App />)
    expect(snap).toMatchSnapshot();
  });

  it('should change the articles displayed after user clicks on a new category', () => {
    const { getByText } = render(<App />)
    const entertainmentButton = getByText('★entertainment')

    fireEvent.click(entertainmentButton)

    expect(getByText('The final season of BoJack Horseman starts October 25th')).toBeInTheDocument();
  });

  it('should change the articles displayed after user searches for a given headline title', () => {
    const { getByPlaceholderText, getByText } = render(<App />)
    const searchInput = getByPlaceholderText('Search for news article here.')
    const searchButton = getByText('Search Now')

    fireEvent.change(searchInput, { target: { value: 'giant chicken' } })
    fireEvent.click(searchButton)

    expect(getByText('Giant Chicken Skeleton In Denver’s Central Library Stands For More Than Just Dinner')).toBeInTheDocument();
  });
});
