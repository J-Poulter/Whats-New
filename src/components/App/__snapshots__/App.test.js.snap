// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render all components needed to match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <div
          class="leftColumn"
        >
          <div>
            <h1>
              What's 
              <span>
                New?
              </span>
            </h1>
            <div>
              <button
                class="menuButton"
              >
                â˜…
                entertainment
              </button>
            </div>
            <div>
              <button
                class="menuButton"
              >
                ðŸ’¡
                local
              </button>
            </div>
            <div>
              <button
                class="menuButton"
              >
                ðŸŽ¦
                health
              </button>
            </div>
            <div>
              <button
                class="menuButton"
              >
                ðŸ”¬
                science
              </button>
            </div>
            <div>
              <button
                class="menuButton"
              >
                ðŸŒ¡
                technology
              </button>
            </div>
          </div>
        </div>
        <div
          class="rightColumn"
        >
          <form>
            <input
              class="searchInput"
              placeholder="Search for news article here."
              value=""
            />
            <button
              class="searchButton"
            >
              Search Now
            </button>
          </form>
          <section>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  The Who postpones Denver Concert at the Pepsi Center
                </h2>
                <p
                  class="articleDescription"
                >
                  Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  Giant Chicken Skeleton In Denverâ€™s Central Library Stands For More Than Just Dinner
                </h2>
                <p
                  class="articleDescription"
                >
                  The 24-foot-tall chicken skeleton towers over the people who pass through the main hall of the Denver Central Library. Skeletal wings extend to the side, and it looks downright prehistoric.
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://www.denverpost.com/wp-content/uploads/2017/06/cd20carsharingdsc_8786.jpg?w=859"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  Car2Go leaving Denver amid company shake-up
                </h2>
                <p
                  class="articleDescription"
                >
                  Car2Go, the popular car-sharing and rental company, will be leaving Denver at the end of the October, the company announced Friday.
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://www.denverpost.com/2019/09/27/car2go-leaving-denver/"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://ewscripps.brightspotcdn.com/dims4/default/4f3b905/2147483647/strip/true/crop/1280x720+0+0/resize/1280x720!/quality/90/?url=https%3A%2F%2Fewscripps.brightspotcdn.com%2Ff2%2F57%2Fcca3119f42c796342f6a8cbf6358%2Fcaden-mural.jpg"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  Mural outside school pays tribute to Denver boy after tragic death
                </h2>
                <p
                  class="articleDescription"
                >
                  A colorful mural outside Ellis Elementary School pays tribute to a little boy who was known for his smile and kindness to other students.
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://www.thedenverchannel.com/news/local-news/mural-outside-school-pays-tribute-to-denver-boy-after-tragic-death"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://localtvkdvr.files.wordpress.com/2019/09/fatal-motorcycle.jpeg?quality=85&strip=all&w=800&h=450&crop=1"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  Motorcyclist killed in crash near Capitol Hill in Denver
                </h2>
                <p
                  class="articleDescription"
                >
                  A motorcyclist was killed in a crash near East 17th Avenue and Park Avenue in Denver early Friday morning.
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://kdvr.com/2019/09/27/motorcyclist-killed-in-crash-near-capitol-hill-in-denver/"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://www.denverpost.com/wp-content/uploads/2019/09/CD10TROYGUARD_AC33486x.jpg?w=781"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  The owner of Los Chingones is planning a food hall, beer garden and brewery in Denver Tech Center
                </h2>
                <p
                  class="articleDescription"
                >
                  Troy Guard, who also owns Guard and Grace, Mister Tuna, Tag and other restaurants, announces a beer and market hall
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://www.denverpost.com/2019/09/27/food-hall-denver-tech-center-troy-guard/"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://theknow.denverpost.com/wp-content/uploads/2019/09/Pokeman2.jpg"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  PokÃ©mon pop-up bar canceled after organizer cites pushback from Nintendo
                </h2>
                <p
                  class="articleDescription"
                >
                  On Wednesday, a publicist for PokÃ©Bar said the Denver stop of the nationally traveling event had been canceled, citing an unspecified directive from PokÃ©mon Nintendo
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://theknow.denverpost.com/2019/09/26/pokebar-pokemon-popup-denver-canceled/225040/"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
            <article
              class="articleCard"
            >
              <img
                alt="Failed to Load"
                class="articleImg"
                src="https://images1.westword.com/imager/u/745xauto/11494355/mason_jar_winter_collins20180216_049.jpg"
              />
              <div
                class="cardInfo"
              >
                <h2
                  class="articleTitle"
                >
                  Medicine Man Dispensaries Halt Sales of Select Vape Products
                </h2>
                <p
                  class="articleDescription"
                >
                  Medicine Man, a Denver-area dispensary chain with four stores across the state, has halted sales of select vaporizer products, specifically pre-filled concentrate cartridges containing propylene glycol or vitamin E acetate.
                </p>
              </div>
              <div
                class="cardLink"
              >
                <a
                  href="https://www.westword.com/marijuana/denver-marijuana-dispensary-pulls-select-vape-products-from-shelves-11494353"
                >
                  Link To Article â†’
                </a>
              </div>
            </article>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <div
        class="leftColumn"
      >
        <div>
          <h1>
            What's 
            <span>
              New?
            </span>
          </h1>
          <div>
            <button
              class="menuButton"
            >
              â˜…
              entertainment
            </button>
          </div>
          <div>
            <button
              class="menuButton"
            >
              ðŸ’¡
              local
            </button>
          </div>
          <div>
            <button
              class="menuButton"
            >
              ðŸŽ¦
              health
            </button>
          </div>
          <div>
            <button
              class="menuButton"
            >
              ðŸ”¬
              science
            </button>
          </div>
          <div>
            <button
              class="menuButton"
            >
              ðŸŒ¡
              technology
            </button>
          </div>
        </div>
      </div>
      <div
        class="rightColumn"
      >
        <form>
          <input
            class="searchInput"
            placeholder="Search for news article here."
            value=""
          />
          <button
            class="searchButton"
          >
            Search Now
          </button>
        </form>
        <section>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                The Who postpones Denver Concert at the Pepsi Center
              </h2>
              <p
                class="articleDescription"
              >
                Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                Giant Chicken Skeleton In Denverâ€™s Central Library Stands For More Than Just Dinner
              </h2>
              <p
                class="articleDescription"
              >
                The 24-foot-tall chicken skeleton towers over the people who pass through the main hall of the Denver Central Library. Skeletal wings extend to the side, and it looks downright prehistoric.
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://www.denverpost.com/wp-content/uploads/2017/06/cd20carsharingdsc_8786.jpg?w=859"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                Car2Go leaving Denver amid company shake-up
              </h2>
              <p
                class="articleDescription"
              >
                Car2Go, the popular car-sharing and rental company, will be leaving Denver at the end of the October, the company announced Friday.
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://www.denverpost.com/2019/09/27/car2go-leaving-denver/"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://ewscripps.brightspotcdn.com/dims4/default/4f3b905/2147483647/strip/true/crop/1280x720+0+0/resize/1280x720!/quality/90/?url=https%3A%2F%2Fewscripps.brightspotcdn.com%2Ff2%2F57%2Fcca3119f42c796342f6a8cbf6358%2Fcaden-mural.jpg"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                Mural outside school pays tribute to Denver boy after tragic death
              </h2>
              <p
                class="articleDescription"
              >
                A colorful mural outside Ellis Elementary School pays tribute to a little boy who was known for his smile and kindness to other students.
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://www.thedenverchannel.com/news/local-news/mural-outside-school-pays-tribute-to-denver-boy-after-tragic-death"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://localtvkdvr.files.wordpress.com/2019/09/fatal-motorcycle.jpeg?quality=85&strip=all&w=800&h=450&crop=1"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                Motorcyclist killed in crash near Capitol Hill in Denver
              </h2>
              <p
                class="articleDescription"
              >
                A motorcyclist was killed in a crash near East 17th Avenue and Park Avenue in Denver early Friday morning.
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://kdvr.com/2019/09/27/motorcyclist-killed-in-crash-near-capitol-hill-in-denver/"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://www.denverpost.com/wp-content/uploads/2019/09/CD10TROYGUARD_AC33486x.jpg?w=781"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                The owner of Los Chingones is planning a food hall, beer garden and brewery in Denver Tech Center
              </h2>
              <p
                class="articleDescription"
              >
                Troy Guard, who also owns Guard and Grace, Mister Tuna, Tag and other restaurants, announces a beer and market hall
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://www.denverpost.com/2019/09/27/food-hall-denver-tech-center-troy-guard/"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://theknow.denverpost.com/wp-content/uploads/2019/09/Pokeman2.jpg"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                PokÃ©mon pop-up bar canceled after organizer cites pushback from Nintendo
              </h2>
              <p
                class="articleDescription"
              >
                On Wednesday, a publicist for PokÃ©Bar said the Denver stop of the nationally traveling event had been canceled, citing an unspecified directive from PokÃ©mon Nintendo
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://theknow.denverpost.com/2019/09/26/pokebar-pokemon-popup-denver-canceled/225040/"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
          <article
            class="articleCard"
          >
            <img
              alt="Failed to Load"
              class="articleImg"
              src="https://images1.westword.com/imager/u/745xauto/11494355/mason_jar_winter_collins20180216_049.jpg"
            />
            <div
              class="cardInfo"
            >
              <h2
                class="articleTitle"
              >
                Medicine Man Dispensaries Halt Sales of Select Vape Products
              </h2>
              <p
                class="articleDescription"
              >
                Medicine Man, a Denver-area dispensary chain with four stores across the state, has halted sales of select vaporizer products, specifically pre-filled concentrate cartridges containing propylene glycol or vitamin E acetate.
              </p>
            </div>
            <div
              class="cardLink"
            >
              <a
                href="https://www.westword.com/marijuana/denver-marijuana-dispensary-pulls-select-vape-products-from-shelves-11494353"
              >
                Link To Article â†’
              </a>
            </div>
          </article>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
